homeassistant/components/tibber/sensor.py

```diff
@@ -384,58 +394,251 @@ def __init__(self, tibber_home: tibber.TibberHome) -> None:
+
+        Returns:
+            List of price data points
+        """
+        if interval == PRICE_INTERVAL_15MIN:
+            return self._get_15min_price_data()
+        return self._get_hourly_price_data()
+
+    def _get_hourly_price_data(self) -> list[dict[str, Any]]:
+        """Get hourly price data from Tibber."""
+        # Get hourly price data
+        hourly_data = getattr(self._tibber_home, "price_info_today", None)
+        if hourly_data and isinstance(hourly_data, list):
+            return hourly_data  # type: ignore[no-any-return]
+
+        # Fallback to current price if no historical data available
+        try:
+            if hasattr(self._tibber_home, "current_price_data"):
+                current_price, _timestamp, level = (
+                    self._tibber_home.current_price_data()
+                )
+                if current_price is not None:
+                    return [
+                        {
+                            "startsAt": dt_util.now().isoformat(),
+                            "total": current_price,
+                            "level": level or "NORMAL",
+                        }
+                    ]
+        except (AttributeError, TypeError, ValueError) as exc:
+            _LOGGER.debug("Failed to get current price data: %s", exc)
+
+        return []
+
+    def _get_15min_price_data(self) -> list[dict[str, Any]]:
+        """Generate 15-minute price data from hourly data."""
+        try:
+            # Get hourly data directly to avoid circular dependency
+            hourly_prices = self._get_hourly_price_data()
+            if not hourly_prices:
+                return []
+
+            fifteen_min_data = []
+            for hour_data in hourly_prices:
+                start_time = dt_util.parse_datetime(hour_data["startsAt"])
+                if start_time is None:
+                    continue
+
+                # Create 4 x 15-minute intervals for each hour
+                for i in range(4):
+                    interval_start = start_time + timedelta(minutes=i * 15)
+                    fifteen_min_data.append(
+                        {
+                            "startsAt": interval_start.isoformat(),
+                            "total": hour_data["total"],
+                            "level": hour_data.get("level", "NORMAL"),
+                        }
+                    )
+
+        except (AttributeError, TypeError, ValueError, KeyError) as err:
+            _LOGGER.warning("Error generating 15-minute price data: %s", err)
+        else:
+            return fifteen_min_data
+        return []
+
     async def async_update(self) -> None:
         """Get the latest data and updates the states."""
+        # Get the selected interval from input_select (if exists)
+        interval = DEFAULT_PRICE_INTERVAL
+        hass = getattr(self, "hass", None)
+        if hass:
+            state = hass.states.get("input_select.tibber_price_interval")
```

Comment:
We don't allow using input_* entities in integrations. The input_* entities are only for users to use.